projects/plugins/jetpack/extensions/blocks/sharing-button/sharing-button.php

```diff
@@ -111,9 +111,16 @@ function sharing_process_requests() {
 
 	// Only process if: single post and share={service} defined
 	if ( ( is_page() || is_single() ) && isset( $_GET['share'] ) && is_string( $_GET['share'] ) ) { // phpcs:ignore WordPress.Security.NonceVerification.Recommended
-		$services     = get_services();
-		$service_name = sanitize_text_field( wp_unslash( $_GET['share'] ) ); // phpcs:ignore WordPress.Security.NonceVerification.Recommended
-		$service      = new $services[ ( $service_name ) ]( $service_name, array() ); // phpcs:ignore WordPress.Security.NonceVerification.Recommended
+		$services                = get_services();
+		$available_service_names = array_keys( $services );
+		$service_name            = sanitize_text_field( wp_unslash( $_GET['share'] ) ); // phpcs:ignore WordPress.Security.NonceVerification.Recommended
+
+		// Only allow services that have been defined in get_services().
+		if ( ! in_array( $service_name, $available_service_names, true ) ) {
```

Comment:
I think we can use `isset( $services[ $service_name ] )` or `array_key_exists( $service_name, $services )` instead of the `in_array` check.

Reply:
We likely need a similar check here https://github.com/Automattic/jetpack/pull/34991/files#diff-afc4158db428ae314674d2fab477ce8ba67bfae66be4987e32e026ebf58952ccL52